syntax = "proto2";

package algotrader.model;

message Bar{
    enum Type{
        Time = 0;
        Tick = 1;
        Volume = 2;
        Dynamic = 3;
    }

    optional int64 inst_id = 1;
    optional Type type = 2;
    optional int32 size = 3;
    optional int32 provider_id = 4;
    optional int64 timestamp = 5;

    optional int64 utc_time = 6;
    optional int64 begin_time = 7;

    optional double open = 9;
    optional double high = 10;
    optional double low = 11;
    optional double close = 12;
    optional double vol = 13;
    optional double adj_close = 14;
    optional double open_int = 15;
}

message Quote{

    optional int64 inst_id = 1;
    optional int32 provider_id = 2;
    optional int64 timestamp = 3;

    optional int64 utc_time = 4;

    optional double bid = 5;
    optional int64 bid_size = 6;
    optional double ask = 7;
    optional int64 ask_size = 8;
}

message Trade{

    optional int64 inst_id = 1;
    optional int32 provider_id = 2;
    optional int64 timestamp = 3;

    optional int64 utc_time = 4;

    optional double price = 5;
    optional int64 size = 6;
}

message MarketDepth{
    enum Side {
        Ask = 0;
        Bid = 1;
    }

    enum Operation {
        Insert = 0;
        Update = 1;
        Delete = 2;
    }

    optional int64 inst_id = 1;
    optional int32 provider_id = 2;
    optional int64 timestamp = 3;

    optional int64 utc_time = 4;

    optional string provider = 5;
    optional int64 position = 6;
    optional Operation operation = 7;
    optional Side side = 8;
    optional double price = 9;
    optional int64 size = 10;
}
